src/main/ArrayVisualizer.java

```diff
@@ -86,6 +110,38 @@ of this software and associated documentation files (the "Software"), to deal
 final public class ArrayVisualizer {
     private static ArrayVisualizer INSTANCE = null;
 
+    private static final int STAT_LINE_BREAK    = 0;
+    private static final int STAT_SORT_IDENTITY = 1;
+    private static final int STAT_ARRAY_LENGTH  = 2;
+    private static final int STAT_FRAMERATE     = 13;
+    private static final int STAT_SORT_DELAY    = 3;
+    private static final int STAT_VISUAL_TIME   = 4;
+    private static final int STAT_EST_SORT_TIME = 5;
+    private static final int STAT_COMPARISONS   = 6;
+    private static final int STAT_SWAPS         = 7;
+    private static final int STAT_REVERSALS     = 8;
+    private static final int STAT_MAIN_WRITE    = 9;
+    private static final int STAT_AUX_WRITE     = 10;
+    private static final int STAT_AUX_ALLOC     = 11;
+    private static final int STAT_SEGMENTS      = 12;
```

Comment:
feels fit for an enum

Reply:
Technically, that could work, but switch statements are slower on enums, and we need inlinable primitive constants for true speed, which we need, as these will be switched on multiple times every frame/

Reply:
is it really significant enough to be a performance bottleneck?

Reply:
Probably not, but...
an enum would
1. Make the code more complex
2. Not more readable 
3. May be a performance bottleneck

Reply:
1. would it?
2. would it make it less readable?
3. drawing text alone is almost certainly much more expensive, and isn't this one of the design goals of enums?

Reply:
1. Yes it would, enums require their own declaration, compile to their own .class file, and are generally just more complex than simple integers.
2. No, but it wouldn't make it more readable either.
3. Yes, and yes, *but* design goals don't mean they're the best for every situation.

Reply:
1. it's like just `enum Blah { BLAH, BLAH; }` and why'd you care about .class files?
2. then it's not really a point either way
3. then this point is dependent on other points

Reply:
So, long story short, my original points, or no points, or whatever. There's literally no real need to make changes to the file past what it is rn.

Reply:
enums mean there's no need to maintain those integers, they don't serve any purpose other than to be unique

Reply:
Ah, but enums use integers. On top of that, enum switch statements are really slow. Each one creates its own class.

Reply:
I mean as in you don't have to write them explicitly, and really? (from reading <https://stackoverflow.com/q/8029040>)

Reply:
I see no reason why the code needs to be *changed* though. If we had gone into this beforehand, maybe I'd have used an enum.

Reply:
it's a pretty simple change though

Reply:
But there's no reason, see my original points. If the runtime cost is negligible, it makes the project bigger, generates an (maybe even two) additional classes (with separate bytecode files) in the background, etc.

Reply:
but we're not trying to optimize every class we can, and if we were, we'd be looking at things other than this

quite simply, I think we should put semantics first

Reply:
If you care so much about making it an enum, why don't *you* do it?